src/App.tsx

```diff
@@ -1,93 +1,262 @@
+} from 'react';
+import {
+  deleteTodos,
+  getActiveTodos, getCompletedTodos, getTodos, patchTodos, postTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Error } from './components/Error/Error';
+import { TodoList } from './components/TodoList';
+import { FilterParams } from './Filter/FilterParams';
+import { Filter } from './Filter';
+import { TodoContext } from './components/TodoContext';
+
+const USER_ID = 9925;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState('');
+  const [isGetDataError, setIsGetDataError] = useState(false);
+  const [isInputEmpty, setIsInputEmpty] = useState(false);
+  const [isPostError, setIsPostError] = useState(false);
+  const [isInputLocked, setIsLockInput] = useState(false);
+  const [isDeleteError, setIsDeleteError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<string | null>(null);
+  const [filterParam, setFilterParam] = useState(
+    localStorage.getItem('filter') || FilterParams.All,
+  );
+  const [activeTodos, setActiveTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [isClearAllCompleted, setIsClearAllCompleted] = useState(false);
+  const [isToggleAllCompleted, setIsToggleAllCompleted] = useState(false);
+  const [isToggleAllActive, setIsToggleAllActive] = useState(false);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      const { completed } = todo;
+
+      switch (filterParam) {
+        case FilterParams.Active:
+          return !completed;
+        case FilterParams.Completed:
+          return completed;
+        default:
+          return true;
+      }
+    });
+  }, [todos, filterParam]);
+
+  useMemo(() => {
+    getActiveTodos(USER_ID)
+      .then(setActiveTodos)
+      .catch(setIsGetDataError);
+
+    getCompletedTodos(USER_ID)
+      .then(setCompletedTodos)
+      .catch(setIsGetDataError);
+  }, [todos, filterParam]);
+
+  const formInputHandler = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.currentTarget;
+
+      setInputValue(value);
+    }, [],
+  );
+
+  const disableErrorHandling = useCallback(() => {
+    setIsGetDataError(false);
+    setIsPostError(false);
+    setIsDeleteError(false);
+    setIsInputEmpty(false);
+  }, []);
+
+  const todosGetter = useCallback(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(setIsGetDataError);
+  }, []);
+
+  useEffect(() => {
+    todosGetter();
+  }, []);
+
+  const postTodoToServer
+  = useCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
+    const { key } = event.nativeEvent;
+
+    if (key === 'Enter') {
+      if (!inputValue.trim()) {
+        setIsInputEmpty(true);
+      } else {
+        setIsLockInput(true);
+        setTempTodo(inputValue);
+
+        postTodos(USER_ID, inputValue)
+          .then(() => todosGetter())
```

Comment:
I noticed that you use 2 different approaches in the project `async/await` and `then` for working with async functions. It's  generally recommended to be consistent and use a single approach throughout a project

Reply:
Fixed